slsim/Sources/source.py

```diff
@@ -125,6 +137,80 @@ def kwargs_variability_extracted(self):
                             "MJD": times,
                             name: magnitudes,
                         }
+
+            # Check if AGN model is used
+            elif any(
+                element in kwargs_variability_list_agn
+                for element in list(self.kwargs_variability)
+            ):
+
+                z = self.source_dict["z"]
+                if self.cosmo is None:
+                    raise ValueError(
+                        "Cosmology cannot be None for AGN class. Please"
+                        "provide a suitable astropy cosmology."
+                    )
+
+                else:
+                    # Get the dict object from the astropy.table.column
+                    agn_kwarg_dict = self.source_dict["kwargs_agn_model"].data[0][0]
+
+                    # Populate "None" for optional keys related to drawing random AGN
+                    random_kwargs = {"random_seed": None, "input_agn_bounds_dict": None}
+                    for opt_key in random_kwargs.keys():
+                        if opt_key in agn_kwarg_dict.keys():
+                            random_kwargs[opt_key] = agn_kwarg_dict[opt_key]
+                            del agn_kwarg_dict[opt_key]
+
+                    # Create the agn object
+                    self.agn_class = agn.RandomAgn(
+                        self.source_dict["i_band_mag"],
```

Comment:
quasar source dict have "ps_mag_i" for magnitude name convention. Here also do "ps_mag_" + band, and band can be passed through kwargs_variability as in supernovae case. kwargs_varibility is already connected through Lens and LensPop class so user can provide any keyword argument for variability model using this dictionary.